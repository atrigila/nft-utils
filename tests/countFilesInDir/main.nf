workflow {
    def trace_timestamp = new java.util.Date().format( 'yyyy-MM-dd_HH-mm-ss')

    ch_stable_content = Channel.of(
    """
    I HAVE STABLE CONTENT
    """.stripIndent().trim())
    .collectFile(storeDir: "${params.outdir}/stable", name: 'stable_content.txt', sort: true, newLine: true)

    ch_stable_name = Channel.of(
    """
    I DO NOT HAVE STABLE CONTENT
    ${trace_timestamp}
    """.stripIndent().trim())
    .collectFile(storeDir: "${params.outdir}/stable", name: 'stable_name.txt', sort: true, newLine: true)

    ch_unstable_name = Channel.of(
    """
    I DO NOT HAVE STABLE NAME
    """.stripIndent().trim())
    .collectFile(storeDir: "${params.outdir}/pipeline_info", name: "execution_trace_${trace_timestamp}.txt", sort: true, newLine: true)

    ch_subdirectory = Channel.of(
    """
    I AM IN A SUBDIRECTORY OF STABLE
    """.stripIndent().trim())
    .collectFile(storeDir: "${params.outdir}/stable/subdirectory", name: "subdir_file.txt", sort: true, newLine: true)

    ch_zip = Channel.of(
    """
    I AM A ZIP FILE
    """.stripIndent().trim())
    .collectFile(storeDir: "${params.outdir}/stable/subdirectory", name: "file.zip", sort: true, newLine: true)
}
