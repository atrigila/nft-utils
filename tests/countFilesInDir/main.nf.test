nextflow_pipeline {

    name "Test countFilesInDir"
    script "./main.nf"
    tag "countFilesInDir"

    test("countFilesInDir") {
            when {
                params {
                    outdir = "$outputDir"
                }
            }

            then {
              // Use countFilesInDir() to get a count of files in the output directory
              def files_in_outdir        = countFilesInDir(params.outdir)
              assert snapshot(
                // Snapshot total files in outdir
                files_in_outdir
              ).match()
            }
        }

    test("countFilesInDir - ignore some files") {
            when {
                params {
                    outdir = "$outputDir"
                }
            }

            then {
              // Use countFilesInDir() to get a count of files in the output directory, except some specific files in a subdirectory
              def stable_name        = countFilesInDir(params.outdir,  ignore: ['pipeline_info/execution_*.{html,txt}'])
                assert snapshot(
                // Only snapshot name
                stable_name,
              ).match()
            }
        }

    test("countFilesInDir - ignore a filetype") {
            when {
                params {
                    outdir = "$outputDir"
                }
            }

            then {
              println(getAllFilesFromDir(params.outdir))
              // Use countFilesInDir() to get a count of files in the output directory, except a specific type
              def stable_name        = countFilesInDir(params.outdir,  ignore: ['**/*.zip'])
                assert snapshot(
                // Only snapshot name
                stable_name,
              ).match()
            }
        }

}
