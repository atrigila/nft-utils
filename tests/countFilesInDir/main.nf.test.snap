{
    "countFilesInDir - ignore a filetype": {
        "content": [
            4
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.0"
        },
        "timestamp": "2024-11-15T21:57:49.331187424"
    },
    "countFilesInDir": {
        "content": [
            5
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.0"
        },
        "timestamp": "2024-11-15T21:37:38.142279268"
    },
    "countFilesInDir - ignore some files": {
        "content": [
            4
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.0"
        },
        "timestamp": "2024-11-15T21:37:43.533042942"
    }
}